Wrappers/Python/test/test_reconstructors.py

```diff
@@ -0,0 +1,286 @@
+# -*- coding: utf-8 -*-
+#   This work is part of the Core Imaging Library (CIL) developed by CCPi 
+#   (Collaborative Computational Project in Tomographic Imaging), with 
+#   substantial contributions by UKRI-STFC and University of Manchester.
+
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from cil.framework import AcquisitionGeometry
+from cil.reconstructors import FBP, Reconstructor
+import unittest
+from scipy.fftpack  import fft, ifft
+import numpy as np
+from utils import has_gpu_tigre, has_ipp
+
+try:
+    from cil.plugins.tigre import ProjectionOperator as ProjectionOperator
+    from cil.plugins.tigre import FBP as FBP_tigre
+    from tigre.utilities.filtering import ramp_flat, filter
+    has_tigre = True
+except ModuleNotFoundError:
+    print(  "These reconstructors require the additional package TIGRE\n" +
+            "Please install it via conda as tigre from the ccpi channel\n"+
+            "Minimal version is 21.01")
+    has_tigre = False
+
+has_tigre = has_tigre and has_gpu_tigre()
+
+has_ipp = has_ipp()
+
+
+class Test_Reconstructor(unittest.TestCase):
+
+    def setUp(self):
+        #%% Setup Geometry
+        voxel_num_xy = 255
+        voxel_num_z = 15
+
+        mag = 2
+        src_to_obj = 50
+        src_to_det = src_to_obj * mag
+
+        pix_size = 0.2
+        det_pix_x = voxel_num_xy
+        det_pix_y = voxel_num_z
+
+        num_projections = 1000
+        angles = np.linspace(0, 360, num=num_projections, endpoint=False)
+
+        self.ag = AcquisitionGeometry.create_Cone2D([0,-src_to_obj],[0,src_to_det-src_to_obj])\
+                                           .set_angles(angles)\
+                                           .set_panel(det_pix_x, pix_size)\
+                                           .set_labels(['angle','horizontal'])
+
+        self.ig = self.ag.get_ImageGeometry()
+
+        self.ag3D = AcquisitionGeometry.create_Cone3D([0,-src_to_obj,0],[0,src_to_det-src_to_obj,0])\
+                                     .set_angles(angles)\
+                                     .set_panel((det_pix_x,det_pix_y), (pix_size,pix_size))\
+                                     .set_labels(['angle','vertical','horizontal'])        
+        self.ig3D = self.ag3D.get_ImageGeometry()
+
+    def test_setup(self):
+        ad3D = self.ag3D.allocate('random')
+        ig3D = self.ag3D.get_ImageGeometry()
+
+
+        reconstructor = Reconstructor(ad3D)
+
+        #test defaults
```

Comment:
It makes sense to split this single test as already implicitly done by the comments.